### ISSUE TITLE
Proxy Target did not return its actual value

### ISSUE THREAD

**Issue by JackCA** [immerjs/immer#537]
## 🙋‍♂ Question

I am starting to receive Proxy errors that looks like:

> `selector.ts:91 Uncaught TypeError: 'get' on proxy: property 'fooObject' is a read-only and non-configurable data property on the proxy target but the proxy did not return its actual value (expected '#<Object>' but got '[object Object]')`

`fooObject` looks like this:
```
{
   items: {}
}
```

Over time, key/values are added to `items` inside of `fooObject` using redux/immer, but `fooObject` itself is never replaced.

The error is thrown inconsistently after an `item` is added/modified and `fooObject` is accessed in a selector.

This *does not* happen with `setAutoFreeze(false)`.

Are there any tips or resources I can use to figure out exactly how/why this is happening? I've dug into the `Proxy` object but I don't see explicitly what the difference is that the error message is throwing.

**_I've been banging my head against the wall for over 2 hours trying to reproduce this in an isolated environment so I apologize for the lack of a repro._**

## Environment

v5.3.6

- **Immer version:**
- [x] Occurs with `setUseProxies(true)`
- [x] Occurs with `setUseProxies(false)` (ES5 only)
- [x] Occurs with `setAutoFreeze(true)`
- [ ] Occurs with `setAutoFreeze(false)`

**Comment by mweststrate** [comment]
Some of the logic of this will be rewritten in the next major. Although this doesn't fix any known issues, it might be worth checking again after it has been released

**Comment by JackCA** [comment]
@mweststrate yeah I did see the v6 alphas but I'll just leave auto-freeze off until you give the go ahead to try those out. Thanks!

**Comment by mweststrate** [comment]
Feel free to retry in 6.0.0 :)
